src/Rules/Arrays/NonexistentOffsetInArrayDimFetchRule.php

```diff
@@ -64,7 +64,7 @@ public function processNode(Node $node, Scope $scope): array
 			return [];
 		}
 
-		if ($scope->isUndefinedExpressionAllowed($node) && !$isOffsetAccessible->no()) {
+		if ($scope->isUndefinedExpressionAllowed($node) && $isOffsetAccessibleType->isOffsetAccessLegal()->yes()) {
```

Comment:
Maybe it's better to add a tip for `$isOffsetAccessibleType->isOffsetAccessLegal()->maybe()`

* The `isOffsetAccessibleType` should be narrowed to a type not including `ObjectType` that do not implement `ArrayAccess::class`
* The class implementing `ArrayAccess::Class` should be marked `final`

Reply:
> The class implementing ArrayAccess::Class should be marked final

I don't get this part. It doesn't matter if a class is final when it already implements `ArrayAccess`. What a finality of such class would achieve?

Reply:
This was just a mistake. I was mixed up with the fact that we cant say `isOffsetAccessLegal()->no()` if the class is not final, because child class might implement ArrayClass.

https://github.com/phpstan/phpstan/issues/10926#issuecomment-2090771360